src/Illuminate/Database/Query/Processors/SqlServerProcessor.php

```diff
@@ -21,13 +21,8 @@ public function processInsertGetId(Builder $query, $sql, $values, $sequence = nu
     {
         $connection = $query->getConnection();
 
-        $connection->insert($sql, $values);
-
-        if ($connection->getConfig('odbc') === true) {
-            $id = $this->processInsertGetIdForOdbc($connection);
-        } else {
-            $id = $connection->getPdo()->lastInsertId();
-        }
+        $result = $connection->selectOne('SET NOCOUNT ON;'.$sql.';SELECT SCOPE_IDENTITY() AS lastInsertedId;', $values, false);
+        $id = is_array($result) ? $result['lastInsertedId'] : $result->lastInsertedId;
```

Comment:
We can also do 
```php
$id = $result->lastInsertedId ?? $result['lastInsertedId'];
```

Reply:
if call $result->lastInsertedId may be having erro on call on none object

Reply:
The current code is fine.

Reply:
yea i was not sure here if i should always expect and object or array.
code was first based on array but it failed because i got an object.